src/Worker.Extensions.DurableTask/HTTP/DurableHttpRequest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask.Http;
+
+/// <summary>
+/// Request used to make an HTTP call through Durable Functions.
+/// </summary>
+public class DurableHttpRequest
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableHttpRequest"/> class.
+    /// </summary>
+    public DurableHttpRequest(
+        HttpMethod method,
+        Uri uri,
+        IDictionary<string, StringValues>? headers = null,
+        string? content = null,
+        bool asynchronousPatternEnabled = true,
+        TimeSpan? timeout = null,
+        HttpRetryOptions? httpRetryOptions = null)
+    {
+        this.Method = method;
+        this.Uri = uri;
+        this.Headers = headers;
+        this.Content = content;
+        this.AsynchronousPatternEnabled = asynchronousPatternEnabled;
+        this.Timeout = timeout;
+        this.HttpRetryOptions = httpRetryOptions;
+    }
+
+    /// <summary>
+    /// HttpMethod used in the HTTP request made by the Durable Function.
+    /// </summary>
+    [JsonPropertyName("method")]
+    [JsonConverter(typeof(HttpMethodConverter))]
+    public HttpMethod Method { get; }
+
+    /// <summary>
+    /// Uri used in the HTTP request made by the Durable Function.
+    /// </summary>
+    [JsonPropertyName("uri")]
+    public Uri Uri { get; }
+
+    /// <summary>
+    /// Headers passed with the HTTP request made by the Durable Function.
+    /// </summary>
+    [JsonPropertyName("headers")]
+    [JsonConverter(typeof(HttpHeadersConverter))]
+    public IDictionary<string, StringValues>? Headers { get; }
+
+    /// <summary>
+    /// Content passed with the HTTP request made by the Durable Function.
+    /// </summary>
+    [JsonPropertyName("content")]
+    public string? Content { get; }
```

Comment:
nit: I typically only put the required properties in the constructor. Something like this can just be a `{ get; set; }`. The only properties that need to be in the constructor are `Method` and `Uri`.